framework/auth/views.py

```diff
@@ -1082,6 +1083,9 @@ def validate_next_url(next_url):
     if next_url.startswith(settings.CAS_SERVER_URL) or next_url.startswith(settings.MFR_SERVER_URL):
         # CAS or MFR
         return True
+    for url in Region.objects.values_list('mfr_url', flat=True):
+        if next_url.startswith(url):
+            return True
```

Comment:
Alternatively, I could do a query to see if the first portion of next_url exists - 
Region.objects.filter(mfr_url__icontains={first-portion-of-next-url})